File: Lot2/Livrable 4/model/Category.php
<?php

//include 'connexion.php';

  class Category{
    public $id;
    public $name;

    
    public function getId()
    {
        return $this->id;
    }

    public function getName()
    {
        return $this->name;
    }

    function __construct($id, $name){
      
      $this->id = $id;
      $this->name=$name;
    }
    
    function __toString(){
        return $this->id . "</br>" .
            $this->name . "</br></br>";
    }
    
    

  }

File: Lot2/Livrable 3/Appli Java (code source)/eCommerce/src/main/java/model/Category.java
package model;

import java.util.ArrayList;

public class Category extends ModelCat {
	public static ArrayList<Category> buffer = new ArrayList<Category>();
	
	public Category(int ID, String name) {
		super(ID, name);
	}
	
	public Category(String name) {
		super(-1, name);
	}
	
	public static void addAll(ArrayList<Category> arrayList) {
		for(int i = 0; i < arrayList.size(); i++) {
			buffer.add(arrayList.get(i));
		}
	}
	
	public static void overwriteBuffer(ArrayList<Category> arrayList) {
		buffer.clear();
		addAll(arrayList);
	}
	
	
	public static Category getCategoryByID(int ID) {
		Category cat, result = null;
		for(int i = 0; i < buffer.size() && result == null; i++) {
			cat = buffer.get(i);
			if(cat.getID() == ID) {
				result = cat;
			}
		}
		
		return result;
	}
	
	public static Category getCategoryByName(String name) {
		Category  cat, result = null;
		for(int i = 0; i < buffer.size() && result == null; i++) {
			cat = buffer.get(i);
			if(cat.getName().equals(name)) {
				result = cat;
			}
		}
		
		return result;
	}
}
